docs/r2/hyperparameter_tuning_with_hparams.ipynb

```diff
@@ -343,169 +283,187 @@
       "outputs": []
     },
     {
+      "cell_type": "markdown",
       "metadata": {
         "id": "46oJF8seXM7v",
         "colab_type": "text"
       },
-      "cell_type": "markdown",
       "source": [
         "For each run, log an hparams summary with the hyperparameters and final accuracy:"
       ]
     },
     {
+      "cell_type": "code",
       "metadata": {
         "id": "8j-fO6nEXRfW",
         "colab_type": "code",
         "colab": {}
       },
-      "cell_type": "code",
       "source": [
         "def run(run_dir, hparams):\n",
-        "  writer = tf.summary.create_file_writer(run_dir)\n",
-        "  summary_start = hparams_summary.session_start_pb(hparams=hparams)\n",
-        "\n",
-        "  with writer.as_default():\n",
+        "  with tf.summary.create_file_writer(run_dir).as_default():\n",
+        "    hp.hparams(hparams)  # record the values used in this trial\n",
         "    accuracy = train_test_model(hparams)\n",
-        "    summary_end = hparams_summary.session_end_pb(api_pb2.STATUS_SUCCESS)\n",
-        "      \n",
-        "    tf.summary.scalar('accuracy', accuracy, step=1, description=\"The accuracy\")\n",
-        "    tf.summary.experimental.write_raw_pb(summary_start.SerializeToString(), step=0)\n",
-        "    tf.summary.experimental.write_raw_pb(summary_end.SerializeToString(), step=0)"
+        "    tf.summary.scalar(METRIC_ACCURACY, accuracy, step=1)"
       ],
       "execution_count": 0,
       "outputs": []
     },
     {
+      "cell_type": "markdown",
       "metadata": {
-        "id": "u2nOgIKAXdcb",
+        "id": "2mYdW0VKLbFE",
         "colab_type": "text"
       },
+      "source": [
+        "When training Keras models, you can use callbacks instead of writing these directly:\n",
```

Comment:
Minor: this code snippet looks a bit awkward in GitHub's notebook rendering, but maybe it's the best option.

Reply:
I’d checked the rendering in the TensorFlow.org staging server and in Colab
before sending the PR. On the docs site, the code block is rendered the
same as other code blocks:

![Screenshot of rendering in docs site](https://user-images.githubusercontent.com/4317806/57351494-ee22fb80-7116-11e9-91b4-f19acb7336a0.png)

In Colab, it’s more similar to GitHub, in that the cell background isn’t
shaded as much as executable code cells:

![Screenshot of rendering in Colab](https://user-images.githubusercontent.com/4317806/57351687-af417580-7117-11e9-9451-dcd0ddc38c48.png)

GitHub, for comparison:

![Screenshot of rendering in GitHub](https://user-images.githubusercontent.com/4317806/57351660-989b1e80-7117-11e9-8bbd-ac53a6f63934.png)

I don’t mind any of the styles, personally. I appreciate the ones that
clearly distinguish executable code from sample-only code (i.e., Colab
and GitHub), but I think that the literal ellipsis is clear enough that
there won’t be any actual confusion.

If you have suggestions or concerns, I’m all ears.

Reply:
It looks good on the TensorFlow.org staging site, so I'm good with this approach. Thank you for exploring it across the different surfaces!